firefox_ui_tests/remote/security/test_safe_browsing_notification.py

```diff
@@ -0,0 +1,120 @@
+
+        # Give the browser a little time, because SafeBrowsing.jsm takes a
+        # while between start up and adding the example urls to the db.
+        # hg.mozilla.org/mozilla-central/file/46aebcd9481e/browser/base/content/browser.js#l1194
+        time.sleep(3)
+
+    def tearDown(self):
+        try:
+            self.browser.tabbar.close_all_tabs([self.browser.tabbar.tabs[0]])
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    def test_notification_bar(self):
+        with self.marionette.using_context('content'):
+            for item in self.test_data:
+                label = item['button_label']
+                report_page, unsafe_page = item['report_page'], item['unsafe_page']
+
+                # Navigate to the unsafe page
+                # Check "ignore warning" link then notification bar's "not badware" button
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_not_badware_button(label, report_page)
+
+                # Return to the unsafe page
+                # Check "ignore warning" link then notification bar's "get me out" button
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_get_me_out_of_here_button()
+
+                # Return to the unsafe page
+                # Check "ignore warning" link then notification bar's "X" button
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_x_button()
+
+    def check_ignore_warning_button(self, unsafe_page):
+        button = self.marionette.find_element(By.ID, 'ignoreWarningButton')
+
+        # Wait for the DOM to receive events
+        time.sleep(1)
+        button.click()
+
+        self.wait_for_condition(expected.element_present(By.ID, 'main-feature'))
+        self.wait_for_condition(expected.element_stale(button))
+        self.assertEquals(self.marionette.get_url(), self.browser.get_loaded_url(unsafe_page))
+
+        # Clean up here since the permission gets set in this function
+        self.utils.remove_perms('www.itisatrap.org', 'safe-browsing')
+
+    # Check the not a forgery or attack button in the notification bar
+    def check_not_badware_button(self, button_label, report_page):
+        with self.marionette.using_context('chrome'):
+            def opener(tab):
+                label = self.browser.get_property(button_label)
+                button = (self.marionette.find_element(By.ID, 'content')
+                          .find_element('anon attribute', {'label': label}))
+                button.click()
+
+            # TODO: better workaround for timeout exceptions (especially  stopbadware.org)
+            self.browser.tabbar.open_tab(opener)
+            self.wait_for_condition(lambda _: report_page in self.browser.navbar.locationbar.value)
```

Comment:
I always see the page begin to load, and I'm not seeing timeout errors today, for whatever that's worth.

Note, the timeout defined here is 5s, not 30s.

We are in chrome context here; marionette.get_url() requires content context.